lib/determinator/cache/fetch_wrapper.rb

```diff
@@ -1,20 +1,49 @@
 module Determinator
   module Cache
     class FetchWrapper
-      # @param cache [#fetch] An instance of a cache class which implements #fetch like ActiveSupport::Cache does
-      def initialize(cache)
-        @cache = cache
+      # @param *caches [#fetch] n instances of an ActiveSupport::Cache (or eqivelent)
+      def initialize(*caches)
+        @caches = caches
       end
 
+      # Call walks through each cache, returning a value if the item exists in
+      # any cache, otherwise popularing each cache with the value of yield.
+
       def call(feature_name)
-        @cache.fetch(key(feature_name)) { yield }
+        value = read_and_upfill(feature_name)
+        # Nil is an acceptable value, do not yield
```

Comment:
Might be worth explaining what a nil value _means_ in this comment? I believe it's a cache hit with no value, right?